geode-core/src/test/java/org/apache/geode/internal/statistics/SampleCollectorTest.java

```diff
@@ -210,7 +210,7 @@ public void testAddHandlerAfterSamples() {
     ResourceInstance resourceInstance = allocatedResourceInstanceInfo.getResourceInstance();
     assertNotNull(resourceInstance);
     assertEquals(0, resourceInstance.getId());
-    assertEquals(0, resourceInstance.getUpdatedStats().length);
+    assertEquals(1, resourceInstance.getUpdatedStats().length);
```

Comment:
Why has this value changed? It's not immediately obvious from the changes why the behaviour here has been affected.

Reply:
In this test case, handler is added after that statistic is updated. According to previous behavior, handler was not notified with the update.